rsmtool/test_utils.py

```diff
@@ -700,3 +706,242 @@ def check_subgroup_outputs(output_dir, experiment_id, subgroups, file_format='cs
             length = len(composition_by_group.loc[composition_by_group['{} set'
                                                                        ''.format(partition)] != 0])
             ok_(length == partition_info.iloc[0][group])
+
+
+class FileUpdater(object):
+
+    """
+    A FileUpdater object is used to update the test outputs
+    for the tests in the `tests_directory` based on the
+    outputs contained in the `updated_outputs_directory`.
+    It does this for all of the experiment tests contained
+    in the test files given by each of the `test_suffixes`.
+
+    Attributes:
+        tests_directory (str): Path to the directory containing the tests whose
+            outputs are to be updated.
+        test_suffixes (list): List of suffixes that will be added to the string
+            "test_experiment_" and located in the `tests_directory` to find
+            the tests that are to be updated.
+        updated_outputs_directory (TYPE): Description
+        deleted_files (list): List of files deleted from `tests directory`.
+        updated_files (list): List of files added or updated for the tests
+            under `tests_directory`.
+        missing_or_empty_sources (list): List of source names whose corresponding
+            directories are either missing under `updated_outputs_directory` or
+            do exist but are empty.
+    """
```

Comment:
Can we change these to `numpy`-style docstrings, rather than Google-style?

```
Attributes
----------
tests_directory : str
    Path to the directory containing the tests whose
    outputs are to be updated.
```

Reply:
Weird, I could have sworn I had set my `AutoDocstring` to numpy and not google since it works fine for methods. I will redo.

Reply:
Ah, it was set to `auto_google`, which means it tries to autodetect the style based on the other docstrings but falls back to google style if it can't.